specs/configure_monitoring.adoc

```diff
@@ -0,0 +1,182 @@
+
+== Problem Description
+
+Tendrl needs to configure collectd for it to start collecting the performance
+statistics at regular intervals of time. This process of configuring collectd
+should require no/minimal user intervention.
+
+== Use Cases
+
+The use cases addressed by this spec are:
+
+* Tendrl core and monitoring stack are installed and user starts node-agent on
+  nodes so that they can be managed by tendrl and the new node needs to be
+  monitored.
+* Tendrl core stack is installed and one/more nodes are managed by tendrl core.
+  Now the user decides to monitor his tendrl managed nodes by installing the
+  tendrl/performance_monitoring.
+
+== Proposed change
+
+* The tendrl/performance_monitoring on start, uploads the default monitoring
+  configuration and flows for generating the collectd configurations.
+* The performance_monitoring application then looks into the etcd's '/nodes'
+  directory to see if there are any nodes that are currently managed by tendrl
+  If there are any, it triggers the monitoring provisioning flow on every node
+  for node monitoring via the core API.
+  ** The provisioning flow for the monitoring stack installs collectd-related
+     pieces on every node and generates the configuration files from templates
+     shipped with the collectd plugins on nodes.
+* For any new node that is managed by tendrl, the performance_monitoring
+  application repeats the above procedure of intiating the monitoring
+  provisioning flow.
+
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+The default monitoring configurations will be a dict of fields like below:
+
+----
+{
+  'interval': 60,
+  'thresholds': {
+    'cpu': {'Warning': 80, 'Failure': 90},
+    'mount_point': {'Warning': 80, 'Failure': 90},
+    'memory': {'Warning': 80, 'Failure': 90},
+    'swap': {'Warning': 50, 'Failure': 70}
+  }
+}
+----
+
+and they will be system wide default configuration that are exposed for
+updates in the form of a yaml file at /etc/tendrl/monitoring_defaults.yaml.
+These values are read at the time of starting of performance_monitoring
+application and hence updates to the file will be reflected on restart of
+performance monitoring application.
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+The flow defined for the configuration is intended to be intially triggered
+internally by the performance_monitoring applciation but for editting of the
+configuration (ex: updating the threshold values to override the defaults),
+the tendrl needs to proxy to the api exposed by the performance_monitoring
+application.
+
+==== Notifications/Monitoring impact:
+
+The performance_monitoring application needs to implement a flow for collectd
+configuration and also this needs to be exposed via an api which the tendrl-api
+proxies to.
+
+==== Tendrl/common impact:
+
+None
+
+==== Tendrl/node_agent impact:
+
+The flow framework needs to be reorganised as described in:
+https://github.com/Tendrl/specifications/pull/8
```

Comment:
No changes in definitions but just that the path to which these definitions will be sent and the way it'll be fetched will change slightly.